### PR TITLE
[FEC-49] - add enum value, localized enum value and field type models

### PR DESCRIPTION
**Pull request by antoniolodias**
#### Summary

This PR creates `enum value`, `localized enum value` and `field type` models needed for this [migration PR](https://github.com/commercetools/merchant-center-frontend/pull/17434) which is based on [batch 12](https://commercetools.atlassian.net/browse/FEC-49) of the [chapter epic](https://commercetools.atlassian.net/browse/FEC-26) where we are migrating tests to use public test-data package

### CODE REVIEW

File: models/type/src/field-type/transformers.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  } as T;
+
+  switch (fields.name) {
+    case 'Enum':
+      return {
+        ...commonFields,
+        values: buildFields(
+          (fields.values as TCustomFieldEnumValue[]) ||
+            list(CustomFieldEnumValue.random),
+          'rest'
```
Comment (jaikumar-tj) [inline models/type/src/field-type/transformers.ts:36]: shouldn't this type `rest` be based on the caller? I mean, this `resolveCommonTransformations` is called by all 3 transformers below

File: models/type/src/field-type/transformers.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import * as CustomFieldLocalizedEnumValue from '../custom-field-localized-enum-value';
+import { TCustomFieldLocalizedEnumValue } from '../custom-field-localized-enum-value';
+import FieldType from './builder';
+import { typeNames, graphqlTypenameByFieldTypeName } from './constants';
+import { TFieldName, TFieldType, TFieldTypeGraphql } from './types';
+
+const list = <T>(fn: (index?: number) => T) =>
+  Array.from({ length: faker.number.int({ min: 1, max: 7 }) }, (_, index) =>
+    fn(index)
+  );
```
Comment (jaikumar-tj) [inline models/type/src/field-type/transformers.ts:19]: Could you maybe add a comment explaining what this function is doing? I see `min` and `max` and index being passed to the callback function, but not fully understand its function.

File: models/type/src/localized-enum-value/types.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { TLocalizedFieldGraphql } from '@commercetools-test-data/commons';
+import type { TBuilder } from '@commercetools-test-data/core';
+
+export type TLocalizedEnumValue = {
+  key: string;
+  label: string;
```
Comment (CarlosCortizasCT) [inline models/type/src/localized-enum-value/types.ts:6]: Could it be that this is of type `LocalizedString` in the REST representation?

File: models/type/src/field-type/transformers.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+const resolveCommonTransformations = <T extends TFieldType | TFieldTypeGraphql>(
+  fields: TFieldType
+): T => {
+  const commonFields = {
+    name: fields.name as TFieldName,
+    __typename: graphqlTypenameByFieldTypeName[fields.name],
```
Comment (CarlosCortizasCT) [inline models/type/src/field-type/transformers.ts:26]: This property should not be common to all transformers, right? I think it's supposed to only exists in the GraphQL representation.

File: models/type/src/field-type/transformers.ts (hunk 1 above)
Comment (CarlosCortizasCT) [reply models/type/src/field-type/transformers.ts]: Updated here: 561d23675c46fbf0307cc2f8c0fa2696220ede83

File: models/type/src/field-type/transformers.ts (hunk 2 above)
Comment (CarlosCortizasCT) [reply models/type/src/field-type/transformers.ts]: Actually, the index was not needed.

This is just used to create an array of a model builder between 1 and 7 elements (what was used before the migration).

Added a comment here: 561d23675c46fbf0307cc2f8c0fa2696220ede83

**Review by CarlosCortizasCT — APPROVED** [review]
Awesome 🚀 

Thanks Antonio 👍

**Review by jaikumar-tj — APPROVED** [review]

**Pull request merged**